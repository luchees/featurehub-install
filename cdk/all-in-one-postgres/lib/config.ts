import { InstanceClass, InstanceSize } from "@aws-cdk/aws-ec2";
import { PostgresEngineVersion } from "@aws-cdk/aws-rds";

interface FeaturehubConfig {
  [k: string]: {
    db: {
      dbName: string;
      userName: string;
      preferredMaintenanceWindow: string;
      version: PostgresEngineVersion;
      instanceClass: InstanceClass;
      instanceSize: InstanceSize;
    };
    fargate: {
      imageVersion: string;
      desiredCount: number;
      maxCapacity: number;
      minCapacity: number;
      cpu: number;
      memoryLimitMiB: number;
    };
  };
}

export const CONFIG: FeaturehubConfig = {
  dev: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T3,
      instanceSize: InstanceSize.MEDIUM,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 2,
      maxCapacity: 8,
      minCapacity: 2,
      cpu: 256,
      memoryLimitMiB: 1024,
    },
  },
  staging: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T2,
      instanceSize: InstanceSize.MICRO,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 1,
      maxCapacity: 2,
      minCapacity: 1,
      cpu: 256,
      memoryLimitMiB: 512,
    },
  },
  production: {
    db: {
      dbName: "featurehub",
      userName: "featurehub",
      preferredMaintenanceWindow: "Wed:10:00-Wed:11:00",
      version: PostgresEngineVersion.VER_13_3,
      instanceClass: InstanceClass.T3,
      instanceSize: InstanceSize.LARGE,
    },
    fargate: {
      imageVersion: "1.4.1",
      desiredCount: 2,
      maxCapacity: 8,
      minCapacity: 2,
      cpu: 256,
      memoryLimitMiB: 1024,
    },
  },
};
