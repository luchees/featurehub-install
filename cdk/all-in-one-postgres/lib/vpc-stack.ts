import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
}

export class VpcStack extends cdk.Stack {
  readonly vpc: ec2.Vpc;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    this.vpc = new ec2.Vpc(this, `${prefix}-VPC`);
  }
}
