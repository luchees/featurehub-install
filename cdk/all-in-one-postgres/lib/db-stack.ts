import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as rds from "@aws-cdk/aws-rds";
import { CONFIG } from "./config";

/**
 * Database Stack for deploying Postgres DB for featurehub
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  vpc: ec2.IVpc;
  dbCredentials: rds.Credentials;
  prefix?: string;
}

export class FeatureHubDatabaseStack extends cdk.Stack {
  readonly database: rds.DatabaseInstance;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    const config = CONFIG[props.stage];

    const database = new rds.DatabaseInstance(this, "FeaturehubDatabase", {
      databaseName: config.db.dbName,
      instanceIdentifier: `${prefix}-featurehub-db`,
      engine: rds.DatabaseInstanceEngine.postgres({
        version: config.db.version,
      }),
      removalPolicy: cdk.RemovalPolicy.RETAIN,
      credentials: props.dbCredentials,
      instanceType: ec2.InstanceType.of(
        config.db.instanceClass,
        config.db.instanceSize
      ),
      vpc: props.vpc,
      vpcSubnets: {
        subnets: props.vpc.isolatedSubnets,
      },
    });
    database.connections.allowFrom(
      ec2.Peer.ipv4(props.vpc.vpcCidrBlock),
      new ec2.Port({
        protocol: ec2.Protocol.TCP,
        stringRepresentation: "Postgres Port",
        fromPort: 5432,
        toPort: 5432,
      }),
      "Allow Access to featureHub database port"
    );

    this.database = database;
  }
}
