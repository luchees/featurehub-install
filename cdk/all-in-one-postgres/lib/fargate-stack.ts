import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as ecs from "@aws-cdk/aws-ecs";
import * as ecs_patterns from "@aws-cdk/aws-ecs-patterns";
import * as iam from "@aws-cdk/aws-iam";
import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
import * as logs from "@aws-cdk/aws-logs";
import * as elb from "@aws-cdk/aws-elasticloadbalancingv2";
import { CONFIG } from "./config";

/**
 * Fargate Stack for deploying featurehub party-server
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
  vpc: ec2.IVpc;
  dbSecret: secretsmanager.Secret;
  dbEndpoint: string;
  dbPort: string;
}

export class FeaturehubFargateStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const config = CONFIG[props.stage];

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;

    const cluster = new ecs.Cluster(this, "featurehubEcsCluster", {
      vpc: props.vpc,
      clusterName: `${prefix}-featurehub-cluster`,
      containerInsights: true,
    });

    const executionRolePolicy = new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      resources: ["*"],
      actions: ["logs:CreateLogStream", "logs:PutLogEvents"],
    });

    const taskDef = new ecs.FargateTaskDefinition(this, "featurehubTaskDef", {
      family: `${prefix}-featurehub-taskdef`,
    });
    taskDef.addToExecutionRolePolicy(executionRolePolicy);
    const loadBalancer = new elb.ApplicationLoadBalancer(
      this,
      "featurehubLoadBalancer",
      {
        vpc: props.vpc,
        loadBalancerName: `${prefix}-featurehub-lb`,
        internetFacing: true,
      }
    );

    const container = taskDef.addContainer(`${prefix}-featurehub`, {
      image: ecs.ContainerImage.fromRegistry(
        `featurehub/party-server:${config.fargate.imageVersion}`
      ),
      logging: new ecs.AwsLogDriver({
        logGroup: new logs.LogGroup(this, "featurehubContainerLogGroup", {
          logGroupName: `/ecs/${prefix}-featurehub`,
        }),
        streamPrefix: `${prefix}-featurehub`,
      }),
      environment: {
        UID: "999",
        GID: "999",
        "DB.URL": `jdbc:postgresql://${props.dbEndpoint}:${props.dbPort}/${config.db.dbName}`,
        "DB.USERNAME": config.db.userName,
        "JERSEY.CORS.HEADERS":
          "X-Requested-With,Authorization,Content-type,Accept-Version,Content-MD5,CSRF-Token,x-ijt,cache-control,x-featurehub,Baggage",
        "REGISTER.URL": `http://${loadBalancer.loadBalancerDnsName}/register-url?token=%s`,
      },
      secrets: {
        "DB.PASSWORD": ecs.Secret.fromSecretsManager(props.dbSecret),
      },
    });

    container.addPortMappings({
      containerPort: 8085,
      protocol: ecs.Protocol.TCP,
    });

    const fargateService = new ecs_patterns.ApplicationLoadBalancedFargateService(
      this,
      "featurehubFargate",
      {
        serviceName: `${prefix}-featurehub-fargate`,
        cluster: cluster,
        platformVersion: ecs.FargatePlatformVersion.VERSION1_4,
        taskDefinition: taskDef,
        healthCheckGracePeriod: cdk.Duration.minutes(5),
        desiredCount: config.fargate.desiredCount,
        loadBalancer,
        cpu: config.fargate.cpu,
        memoryLimitMiB: config.fargate.memoryLimitMiB,
      }
    );

    fargateService.targetGroup.configureHealthCheck({
      enabled: true,
      path: "/metric/health/readyness",
      port: "8085",
      healthyHttpCodes: "200",
    });

    //Cookie stickiness needs to be enabled to make sure caching works
    fargateService.targetGroup.enableCookieStickiness(cdk.Duration.days(1));

    const scaling = fargateService.service.autoScaleTaskCount({
      maxCapacity: config.fargate.maxCapacity,
      minCapacity: config.fargate.minCapacity,
    });
    scaling.scaleOnCpuUtilization("featurehubCpuScaling", {
      policyName: `${prefix}-featurehub-cpu-scaling`,
      targetUtilizationPercent: 80,
      scaleInCooldown: cdk.Duration.seconds(60),
      scaleOutCooldown: cdk.Duration.seconds(60),
    });
    scaling.scaleOnMemoryUtilization("featurehubMemScaling", {
      policyName: `${prefix}-featurehub-mem-scaling`,
      targetUtilizationPercent: 80,
      scaleInCooldown: cdk.Duration.seconds(60),
      scaleOutCooldown: cdk.Duration.seconds(60),
    });
  }
}
