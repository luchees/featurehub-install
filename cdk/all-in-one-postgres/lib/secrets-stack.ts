import * as cdk from "@aws-cdk/core";
import * as rds from "@aws-cdk/aws-rds";
import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
import { CONFIG } from "./config";

/**
 * Secrets Stack for creating the Secrets to be used with Featurehub Database
 */

interface ExtStackProps extends cdk.StackProps {
  stage: string;
  prefix?: string;
}

export class FeatureHubSecretsStack extends cdk.Stack {
  readonly dbPasswordSecret: secretsmanager.Secret;
  readonly dbCredentials: rds.Credentials;

  constructor(scope: cdk.Construct, id: string, props: ExtStackProps) {
    super(scope, id, props);

    const prefix = props.prefix
      ? `${props.prefix}-${props.stage}`
      : props.stage;
    const config = CONFIG[props.stage];

    const dbPasswordSecret = new secretsmanager.Secret(
      this,
      "FeaturehubDbPassword",
      {
        secretName: `${prefix}-featurehub-db-password`,
        generateSecretString: {
          excludeCharacters: "%+~`#$&*()|[]{}:;<>?!/@ \\\"'", //password characters not supported by postgresql
        },
      }
    );
    // RDS Credentials to be used by RDS
    const dbCredentials = rds.Credentials.fromPassword(
      config.db.userName,
      dbPasswordSecret.secretValue
    );

    this.dbPasswordSecret = dbPasswordSecret;
    this.dbCredentials = dbCredentials;
  }
}
