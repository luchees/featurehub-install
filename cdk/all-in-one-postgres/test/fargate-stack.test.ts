import { SynthUtils } from "@aws-cdk/assert";
import { Vpc } from "@aws-cdk/aws-ec2";
import { Secret } from "@aws-cdk/aws-secretsmanager";
import * as cdk from "@aws-cdk/core";
import * as Cdk from "../lib/fargate-stack";

describe("Featurehub fargate Stack", () => {
  test("snapshot", () => {
    const app = new cdk.App();
    const cdkStack = new cdk.Stack(app, "MyStack");
    const stack = new Cdk.FeaturehubFargateStack(app, "MyFargateStack", {
      stage: "dev",
      vpc: new Vpc(cdkStack, "MyVpc"),
      dbEndpoint: "featurehub.database.com",
      dbPort: "8099",
      dbSecret: new Secret(cdkStack, "MySecret"),
    });
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});
