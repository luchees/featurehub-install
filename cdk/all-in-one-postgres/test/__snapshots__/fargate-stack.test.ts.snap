// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featurehub fargate Stack snapshot 1`] = `
Object {
  "Outputs": Object {
    "featurehubFargateLoadBalancerDNS9CBC8007": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "featurehubLoadBalancerDFF2FDB0",
          "DNSName",
        ],
      },
    },
    "featurehubFargateServiceURLA7F7E567": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "http://",
            Object {
              "Fn::GetAtt": Array [
                "featurehubLoadBalancerDFF2FDB0",
                "DNSName",
              ],
            },
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "featurehubContainerLogGroup8540B674": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": "/ecs/dev-featurehub",
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "featurehubEcsClusterF335BC98": Object {
      "Properties": Object {
        "ClusterName": "dev-featurehub-cluster",
        "ClusterSettings": Array [
          Object {
            "Name": "containerInsights",
            "Value": "enabled",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "featurehubFargateServiceA22EA68F": Object {
      "DependsOn": Array [
        "featurehubLoadBalancerPublicListenerECSGroup09533230",
        "featurehubLoadBalancerPublicListenerB6F0D687",
      ],
      "Properties": Object {
        "Cluster": Object {
          "Ref": "featurehubEcsClusterF335BC98",
        },
        "DeploymentConfiguration": Object {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 50,
        },
        "DesiredCount": 2,
        "EnableECSManagedTags": false,
        "HealthCheckGracePeriodSeconds": 300,
        "LaunchType": "FARGATE",
        "LoadBalancers": Array [
          Object {
            "ContainerName": "dev-featurehub",
            "ContainerPort": 8085,
            "TargetGroupArn": Object {
              "Ref": "featurehubLoadBalancerPublicListenerECSGroup09533230",
            },
          },
        ],
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "DISABLED",
            "SecurityGroups": Array [
              Object {
                "Fn::GetAtt": Array [
                  "featurehubFargateServiceSecurityGroupC773E7F0",
                  "GroupId",
                ],
              },
            ],
            "Subnets": Array [
              Object {
                "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcPrivateSubnet1Subnet5057CF7EBB23EA1E",
              },
              Object {
                "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcPrivateSubnet2Subnet0040C983A26C73F6",
              },
            ],
          },
        },
        "PlatformVersion": "1.4.0",
        "ServiceName": "dev-featurehub-fargate",
        "TaskDefinition": Object {
          "Ref": "featurehubTaskDef353B2C82",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "featurehubFargateServiceSecurityGroupC773E7F0": Object {
      "Properties": Object {
        "GroupDescription": "MyFargateStack/featurehubFargate/Service/SecurityGroup",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcF9F0CA6FBC8737E9",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "featurehubFargateServiceSecurityGroupfromMyFargateStackfeaturehubLoadBalancerSecurityGroup59D0DFCB8085DAE252DC": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 8085,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "featurehubFargateServiceSecurityGroupC773E7F0",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "featurehubLoadBalancerSecurityGroup35E8B780",
            "GroupId",
          ],
        },
        "ToPort": 8085,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "featurehubFargateServiceTaskCountTargetD855076C": Object {
      "Properties": Object {
        "MaxCapacity": 8,
        "MinCapacity": 2,
        "ResourceId": Object {
          "Fn::Join": Array [
            "",
            Array [
              "service/",
              Object {
                "Ref": "featurehubEcsClusterF335BC98",
              },
              "/",
              Object {
                "Fn::GetAtt": Array [
                  "featurehubFargateServiceA22EA68F",
                  "Name",
                ],
              },
            ],
          ],
        },
        "RoleARN": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":role/aws-service-role/ecs.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_ECSService",
            ],
          ],
        },
        "ScalableDimension": "ecs:service:DesiredCount",
        "ServiceNamespace": "ecs",
      },
      "Type": "AWS::ApplicationAutoScaling::ScalableTarget",
    },
    "featurehubFargateServiceTaskCountTargetfeaturehubCpuScaling1391631C": Object {
      "Properties": Object {
        "PolicyName": "dev-featurehub-cpu-scaling",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "featurehubFargateServiceTaskCountTargetD855076C",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageCPUUtilization",
          },
          "ScaleInCooldown": 60,
          "ScaleOutCooldown": 60,
          "TargetValue": 80,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "featurehubFargateServiceTaskCountTargetfeaturehubMemScalingA551CA28": Object {
      "Properties": Object {
        "PolicyName": "dev-featurehub-mem-scaling",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "featurehubFargateServiceTaskCountTargetD855076C",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageMemoryUtilization",
          },
          "ScaleInCooldown": 60,
          "ScaleOutCooldown": 60,
          "TargetValue": 80,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "featurehubLoadBalancerDFF2FDB0": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "false",
          },
        ],
        "Name": "dev-featurehub-lb",
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "featurehubLoadBalancerSecurityGroup35E8B780",
              "GroupId",
            ],
          },
        ],
        "Subnets": Array [
          Object {
            "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcPublicSubnet1SubnetF6608456AB60A910",
          },
          Object {
            "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcPublicSubnet2Subnet492B6BFBA5DEC6B5",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "featurehubLoadBalancerPublicListenerB6F0D687": Object {
      "Properties": Object {
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "featurehubLoadBalancerPublicListenerECSGroup09533230",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "featurehubLoadBalancerDFF2FDB0",
        },
        "Port": 80,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "featurehubLoadBalancerPublicListenerECSGroup09533230": Object {
      "Properties": Object {
        "HealthCheckEnabled": true,
        "HealthCheckPath": "/metric/health/readyness",
        "HealthCheckPort": "8085",
        "Matcher": Object {
          "HttpCode": "200",
        },
        "Port": 80,
        "Protocol": "HTTP",
        "TargetGroupAttributes": Array [
          Object {
            "Key": "stickiness.enabled",
            "Value": "true",
          },
          Object {
            "Key": "stickiness.type",
            "Value": "lb_cookie",
          },
          Object {
            "Key": "stickiness.lb_cookie.duration_seconds",
            "Value": "86400",
          },
        ],
        "TargetType": "ip",
        "VpcId": Object {
          "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcF9F0CA6FBC8737E9",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "featurehubLoadBalancerSecurityGroup35E8B780": Object {
      "Properties": Object {
        "GroupDescription": "Automatically created Security Group for ELB MyFargateStackfeaturehubLoadBalancer9951DB01",
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 80",
            "FromPort": 80,
            "IpProtocol": "tcp",
            "ToPort": 80,
          },
        ],
        "VpcId": Object {
          "Fn::ImportValue": "MyStack:ExportsOutputRefMyVpcF9F0CA6FBC8737E9",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "featurehubLoadBalancerSecurityGrouptoMyFargateStackfeaturehubFargateServiceSecurityGroup7B9AEC198085CDD2A178": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "featurehubFargateServiceSecurityGroupC773E7F0",
            "GroupId",
          ],
        },
        "FromPort": 8085,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "featurehubLoadBalancerSecurityGroup35E8B780",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 8085,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "featurehubTaskDef353B2C82": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "UID",
                "Value": "999",
              },
              Object {
                "Name": "GID",
                "Value": "999",
              },
              Object {
                "Name": "DB.URL",
                "Value": "jdbc:postgresql://featurehub.database.com:8099/featurehub",
              },
              Object {
                "Name": "DB.USERNAME",
                "Value": "featurehub",
              },
              Object {
                "Name": "JERSEY.CORS.HEADERS",
                "Value": "X-Requested-With,Authorization,Content-type,Accept-Version,Content-MD5,CSRF-Token,x-ijt,cache-control,x-featurehub,Baggage",
              },
              Object {
                "Name": "REGISTER.URL",
                "Value": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "http://",
                      Object {
                        "Fn::GetAtt": Array [
                          "featurehubLoadBalancerDFF2FDB0",
                          "DNSName",
                        ],
                      },
                      "/register-url?token=%s",
                    ],
                  ],
                },
              },
            ],
            "Essential": true,
            "Image": "featurehub/party-server:1.4.1",
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "featurehubContainerLogGroup8540B674",
                },
                "awslogs-region": Object {
                  "Ref": "AWS::Region",
                },
                "awslogs-stream-prefix": "dev-featurehub",
              },
            },
            "Name": "dev-featurehub",
            "PortMappings": Array [
              Object {
                "ContainerPort": 8085,
                "Protocol": "tcp",
              },
            ],
            "Secrets": Array [
              Object {
                "Name": "DB.PASSWORD",
                "ValueFrom": Object {
                  "Fn::ImportValue": "MyStack:ExportsOutputRefMySecret8FE80B5183540F68",
                },
              },
            ],
          },
        ],
        "Cpu": "256",
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "featurehubTaskDefExecutionRoleF272870F",
            "Arn",
          ],
        },
        "Family": "dev-featurehub-taskdef",
        "Memory": "512",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "featurehubTaskDefTaskRole3E05763E",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "featurehubTaskDefExecutionRoleDefaultPolicyC8957E70": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "featurehubContainerLogGroup8540B674",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::ImportValue": "MyStack:ExportsOutputRefMySecret8FE80B5183540F68",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "featurehubTaskDefExecutionRoleDefaultPolicyC8957E70",
        "Roles": Array [
          Object {
            "Ref": "featurehubTaskDefExecutionRoleF272870F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "featurehubTaskDefExecutionRoleF272870F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "featurehubTaskDefTaskRole3E05763E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
